### PR TITLE
[WIP] update GitHub tests workflow

### PR DESCRIPTION
**Pull request by r-f-g**
- [comment] add "recheck-snap" trigger
 - [comment] add "recheck-full" trigger
 - [review] add "APPROVED" trigger
 - TODO: check if test need to run again
 - ~~TODO: join tests from pull_request and pull_request_review events~~ -> tests will be skipped (see above)
 - TODO: trigger only on author or contributor comment

### CODE REVIEW

**Comment by r-f-g** [comment]
recheck-snap

**Comment by r-f-g** [comment]
recheck-snap

**Comment by r-f-g** [comment]
recheck-snap

**Comment by r-f-g** [comment]
recheck-snap

**Comment by r-f-g** [comment]
recheck-snap

**Review by r-f-g — COMMENTED** [review]
recheck-snap

**Review by mkalcok — APPROVED** [review]

**Comment by r-f-g** [comment]
recheck-snap

**Comment by r-f-g** [comment]
recheck-snap

**Review by r-f-g — COMMENTED** [review]
recheck-snap

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -4,12 +4,43 @@ name: Test suite
 
 on:
   pull_request:
+    types: [ opened, synchronize, reopened, ready_for_review ]
     branches: [ master ]
 
+  pull_request_review:
+    types: [ submitted ]
+
+  pull_request_review_comment:
```
Comment (r-f-g) [inline .github/workflows/tests.yml:13]: TEST

**Review by r-f-g — COMMENTED** [review]
recheck-snap

**Review by r-f-g — COMMENTED** [review]
recheck-snap

**Review by r-f-g — COMMENTED** [review]
recheck-snap

**Review by r-f-g — COMMENTED** [review]
recheck-snap

**Review by r-f-g — COMMENTED** [review]
recheck-snap

**Review by r-f-g — COMMENTED** [review]
recheck-snap

**Pull request closed without merging**